File: java/src/sample/replicant/sample2/MockRequestQueue.java
// Copyright 2009 Kiel Hodges
package replicant.sample2;

import replicant.*;
import static replicant.Replicant.*;

public class MockRequestQueue implements RequestQueue {

  public MockRequestQueue(ExpectationEnforcer expectationEnforcer) {
    nextRequest = methodMocker(this, "nextRequest").
      orderingResponses().
      enforcingExpectationsWith(expectationEnforcer).
      with0Args();
  }

  public Request nextRequest() {
    return nextRequest.call();
  }
  
  public final MethodMocker0<Request> nextRequest;
  
}

File: scala/src/test/scala/sample2/MockRequestQueue.scala
// Copyright 2011 Kiel Hodges
package sample2

import replicant._
import experiment1._

class MockRequestQueue extends XMockController[RequestQueue] { self =>

  val testDouble: RequestQueue = new RequestQueue {
    def nextRequest: Option[Request] = self.nextRequest.response
  }

  val nextRequest = method("nextRequest", testDouble.nextRequest _)
  
}
